packages/core/src/browser/quick-open/quick-open-action.ts

```diff
@@ -0,0 +1,116 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Disposable } from '../../common/disposable';
+import { injectable, unmanaged } from 'inversify';
+
+export interface QuickOpenAction extends Disposable {
+    id: string;
+    label: string;
+    tooltip: string;
+    class: string | undefined;
+    enabled: boolean;
+    checked: boolean;
+    radio: boolean;
+    // tslint:disable-next-line:no-any
+    run(event?: any): PromiseLike<any>;
+}
+
+export interface QuickOpenActionProvider {
+    // tslint:disable-next-line:no-any
+    hasActions(item: any): boolean;
+    // tslint:disable-next-line:no-any
+    getActions(item: any): Promise<QuickOpenAction[]>;
+}
+
+@injectable()
+export abstract class QuickOpenBaseAction implements QuickOpenAction {
+    protected actionId: string;
+    protected actionLabel: string;
+    protected actionTooltip: string;
+    protected actionCssClass: string | undefined;
+    protected actionEnabled: boolean;
+    protected actionChecked: boolean;
+    protected actionRadio: boolean;
+
+    constructor(
+        @unmanaged() id: string,
```

Comment:
you could use `Options` pattern like for `QuickOpenItem`: https://github.com/theia-ide/theia/blob/7edb5f2e9a94272360404d0f48f9a1b88989e9b1/packages/core/src/browser/quick-open/quick-open-model.ts#L54

Reply:
Done